core/headerchain.go

```diff
@@ -307,6 +307,33 @@ func (hc *HeaderChain) GetBlockHashesFromHash(hash common.Hash, max uint64) []co
 	return chain
 }
 
+// GetAncestor retrieves the Nth ancestor of a given block. It assumes that either the given block or
+// a close ancestor of it is canonical. maxNonCanonical points to a downwards counter limiting the
+// number of blocks to be individually checked before we reach the canonical chain.
+func (hc *HeaderChain) GetAncestor(hash common.Hash, number, ancestor uint64, maxNonCanonical *uint64) (common.Hash, uint64) {
```

Comment:
Could you expand the docs to clarify whether `ancestor` is a block number, or a counter for how many blocks away from `number` we want? 
That is: does `ancestor = 1` mean the parent of `number`?

Reply:
"That is: does ancestor = 1 mean the parent of number?" yes, added clarification to the docs.